### PR TITLE
Implement OLM operator tests on PRs

### PR DESCRIPTION
**Pull request by flacatus**
### Test OLM files
This PR implement the necesary setup scrips to automate the test of OLM files and especially OLM operator:
1. Build the docker images of dedicated catalog for Openshift and Kubernetes and then run locally to test.
2. Deploy localy the operator using previous docker images created.

Reference issue
https://github.com/eclipse/che/issues/15803

### CODE REVIEW

**Comment by flacatus** [comment]
[test]

**Comment by flacatus** [comment]
cc @AndrienkoAleksandr @davidfestal

**Comment by tolusha** [comment]
Let's wait until 7.9.0 is released.

**Comment by flacatus** [comment]
Cool!

File: olm/olm.sh [hunk 1]
```diff
@@ -59,12 +62,31 @@ then
...
+kind: CatalogSource
+metadata:
+  name: ${packageName}
+  namespace: ${namespace}
+spec:
+  sourceType: grpc
+  image: ${CATALOG_IMAGENAME}
+EOF
+    else
+    cat ${platformPath}/operator-source.yaml
```
Comment (AndrienkoAleksandr) [inline olm/olm.sh]: Align code format here, please.

File: olm/olm.sh [hunk 2]
```diff
@@ -59,12 +62,31 @@ then
   exit 1
 fi
 
+catalog_source() {
+  echo "--- Use default eclipse che application registry ---"
+  echo $SOURCE_INSTALL
+    if [ ${SOURCE_INSTALL} == "LocalCatalog" ]; then
+    marketplaceNamespace=${namespace};
```
Comment (AndrienkoAleksandr) [inline olm/olm.sh]: Align code format here, please.

File: .ci/cico_olm_prcheck.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  installStartDocker
+  minikube_installation
+}
+
+run_olm_tests() {
+  printInfo "Starting minikube VM to test kubernetes olm files..."
+  minikube start --memory=${MINIKUBE_MEMORY}
+  minikube addons enable ingress
+  printInfo "Running olm files tests on Kubernetes"
+  sh ${OPERATOR_REPO}/olm/testCatalogSource.sh kubernetes stable ${NAMESPACE}
```
Comment (AndrienkoAleksandr) [inline .ci/cico_olm_prcheck.sh]: Sorry,  script has name cico_olm_prcheck.sh , so I thought it should use nightly channel, but maybe I miss something...

Comment (flacatus) [reply .ci/cico_olm_prcheck.sh]: Hmm, Good point! Do you think could be a good Idea to test both channels?

Comment (AndrienkoAleksandr) [reply .ci/cico_olm_prcheck.sh]: @tolusha ?

File: .ci/util/ci_common.sh [hunk 4]
```diff
@@ -67,3 +86,29 @@ generate_self_signed_certs() {
...
+  && chmod +x minikube
+  sudo install minikube /usr/local/bin/ && rm -rf minikube
+}
+
+installEpelRelease() {
+  if yum repolist | grep epel; then
+    printWarn "Epel already installed, skipping instalation."
+  else
+    #excluding mirror1.ci.centos.org 
+    echo "exclude=mirror1.ci.centos.org" >> /etc/yum/pluginconf.d/fastestmirror.conf
```
Comment (rhopp) [inline .ci/util/ci_common.sh]: Why are you excluding this mirror here?

Comment (flacatus) [reply .ci/util/ci_common.sh]: I discovered few days ago an issues in VM with epel. https://bugs.centos.org/view.php?id=16337

File: .ci/cico_olm_prcheck.sh (hunk 3 above)
Comment (tolusha) [reply .ci/cico_olm_prcheck.sh]: @AndrienkoAleksandr 
nice point, should be nightly

File: .ci/util/ci_common.sh (hunk 4 above)
Comment (rhopp) [reply .ci/util/ci_common.sh]: Seems like Vipul already closed that issue (few days ago). Can you confirm this is still happening?

Comment (flacatus) [reply .ci/util/ci_common.sh]: Hmm, In my Vagrant machine seems the issue is solved.

File: olm/olm.sh [hunk 5]
```diff
@@ -76,23 +98,34 @@ applyCheOperatorSource() {
...
+      kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/03_operatorsource.crd.yaml
+      kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/04_service_account.yaml
+      kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/05_role.yaml
+      kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/06_role_binding.yaml
+      sleep 1
+      kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/07_upstream_operatorsource.cr.yaml
+      kubectl apply -f https://raw.githubusercontent.com/operator-framework/operator-marketplace/master/deploy/upstream/08_operator.yaml
+
+      kubectl get catalogsource/"${packageName}" -n "${marketplaceNamespace}" -o json | jq '.metadata.namespace = "olm" | del(.metadata.creationTimestamp) | del(.metadata.uid) | del(.metadata.resourceVersion) | del(.metadata.generation) | del(.metadata.selfLink) | del(.status)' | kubectl apply -f -
```
Comment (davidfestal) [inline olm/olm.sh]: I don't understand what you're trying to achieve here.

File: olm/olm.sh [hunk 6]
```diff
@@ -112,30 +145,20 @@ installOperatorMarketPlace() {
       echo "Catalog source not created after 2 minutes"
       exit 1
     fi
-
-    kubectl get catalogsource/"${packageName}" -n "${marketplaceNamespace}" -o json | jq '.metadata.namespace = "olm" | del(.metadata.creationTimestamp) | del(.metadata.uid) | del(.metadata.resourceVersion) | del(.metadata.generation) | del(.metadata.selfLink) | del(.status)' | kubectl apply -f -
```
Comment (davidfestal) [inline olm/olm.sh]: Why did you move these instructions at a place where `applyCheOperatorSource` hasn't been applied already ?

File: olm/olm.sh [hunk 7]
```diff
@@ -112,30 +145,20 @@ installOperatorMarketPlace() {
...
+  echo "Subscribing to version: ${CSV}"
+
   kubectl apply -f - <<EOF
-apiVersion: v1
-kind: Namespace
-metadata:
-  name: ${namespace}
----
-apiVersion: operators.coreos.com/v1alpha2
+apiVersion: operators.coreos.com/v1
```
Comment (davidfestal) [inline olm/olm.sh]: Is this change from `v1alpha2` to `v1` OK for any platform ? I assume it is for the last Minikube, but what about OpenShift 4.2 or other platforms ? Are there constraints that require us changing this in this PR ?

Comment (flacatus) [reply olm/olm.sh]: In fact I had problem in Openshift 4.3.1 with this version. With **v1** I don't have any problem in minikube or Openshift 4.x

Comment (davidfestal) [reply olm/olm.sh]: OK

File: olm/olm.sh (hunk 6 above)
Comment (flacatus) [reply olm/olm.sh]: @davidfestal Sorry was an error from my side.

File: olm/olm.sh (hunk 5 above)
Comment (flacatus) [reply olm/olm.sh]: I tried to differentiate the installation from marketplace catalog or from local catalog. But I studied the code again and is not necessary this line.
